crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -631,19 +631,45 @@ impl IndexerReader {
         let digests = digests
             .iter()
             .map(|digest| digest.inner().to_vec())
-            .collect::<Vec<_>>();
-        run_query!(&self.pool, |conn| {
+            .collect::<HashSet<_>>();
+        let checkpointed_txs = run_query!(&self.pool, |conn| {
             transactions::table
                 .inner_join(
                     tx_digests::table
                         .on(transactions::tx_sequence_number.eq(tx_digests::tx_sequence_number)),
                 )
                 // we filter the tx_digests table because it is indexed by digest,
                 // transactions table is not
-                .filter(tx_digests::tx_digest.eq_any(digests))
+                .filter(tx_digests::tx_digest.eq_any(&digests))
                 .select(StoredTransaction::as_select())
                 .load::<StoredTransaction>(conn)
-        })
+        })?;
+        if checkpointed_txs.len() == digests.len() {
+            return Ok(checkpointed_txs);
+        }
+        let missing_digests = {
+            let mut missing_digests = digests;
+            for tx in &checkpointed_txs {
+                missing_digests.remove(&tx.transaction_digest);
+            }
+            missing_digests
+        };
```

Comment:
Nit: unnecessary block
```suggestion
        let mut missing_digests = digests;
        for tx in &checkpointed_txs {
            missing_digests.remove(&tx.transaction_digest);
        }
```

Reply:
Done